File: coding/22nd October 2025/Majority Element.java
//Majority Element
class Solution 
{
    public int majorityElement(int[] nums) 
    {
        int n = nums.length;
        HashMap<Integer, Integer> map = new HashMap<Integer,Integer>();
        for(int i = 0; i < n; i++)
        {
            if(map.containsKey(nums[i]))
                map.put(nums[i], map.get(nums[i]) + 1);
            else
                map.put(nums[i], 1);
        }
        for(int i = 0; i < n; i++)
            if(map.get(nums[i]) > n/2)
                return nums[i];
        return 0;
    }
}

File: Coding in Python/11th January 2026/Majority Element.py
#Majority Element
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        nums.sort()
        c = 1
        for i in range(1, len(nums)):
            if nums[i] == nums[i - 1]:
                c += 1
                if c > (len(nums) // 2):
                    return nums[i]
            else:
                c = 1
        return nums[0]
